scvi/module/_mixupvae.py

```diff
@@ -336,16 +346,20 @@ def inference(
                     # if n_cat == 0 then no batch index was given, so skip it
                     one_hot_cat_covs = one_hot(cat_covs[j], n_cat)
                     categorical_input.append(one_hot_cat_covs)
-                    ont_hot_cat_covs_pure = one_hot_cat_covs.mean(axis=0).resize(1,n_cat)
-                    categorical_pseudobulk_input.append(ont_hot_cat_covs_pure)
+                    one_hot_cat_covs_pseudobulk = one_hot_cat_covs.mean(axis=0).resize(1,n_cat)
+                    categorical_pseudobulk_input.append(one_hot_cat_covs_pseudobulk)
                     categorical_signature_input_temp = []
                     for cell_type in unique_indices:
                         idx = (y == cell_type).flatten()
                         one_hot_cat_covs_pure = one_hot_cat_covs[idx, :].mean(axis=0)
                         categorical_signature_input_temp.append(one_hot_cat_covs_pure)
                     categorical_signature_input.append(torch.stack(categorical_signature_input_temp))
                     j+=1
-        else:
+            categorical_input_bis = copy.copy(categorical_input)
```

Comment:
Is it different from copy.deepcopy ?

Reply:
Yes it's different - > https://docs.python.org/3/library/copy.html

We might want to use a `deepcopy`, since the original object can be updated